### ISSUE TITLE
scilab_kernel Exception

### ISSUE THREAD

**Issue by lc82111** [Calysto/scilab_kernel#2]
[IPKernelApp] ERROR | Exception in message handler:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/ipykernel/kernelbase.py", line 228, in dispatch_shell
    handler(stream, idents, msg)
  File "/usr/local/lib/python2.7/dist-packages/ipykernel/kernelbase.py", line 390, in execute_request
    user_expressions, allow_stdin)
  File "/usr/local/lib/python2.7/dist-packages/metakernel/_metakernel.py", line 350, in do_execute
    retval = self.do_execute_direct(code)
  File "/usr/local/lib/python2.7/dist-packages/scilab_kernel/kernel.py", line 105, in do_execute_direct
    self.handle_plot_settings()
  File "/usr/local/lib/python2.7/dist-packages/scilab_kernel/kernel.py", line 189, in handle_plot_settings
    super(ScilabKernel, self).do_execute_direct('\n'.join(cmds))
  File "/usr/local/lib/python2.7/dist-packages/metakernel/process_metakernel.py", line 79, in do_execute_direct
    stdin_handler=self.raw_input)
  File "/usr/local/lib/python2.7/dist-packages/metakernel/replwrap.py", line 164, in run_command
    self._expect_prompt(timeout=timeout)
  File "/usr/local/lib/python2.7/dist-packages/metakernel/replwrap.py", line 114, in _expect_prompt
    pos = self._expect_inner(expects, timeout)
  File "/usr/local/lib/python2.7/dist-packages/metakernel/replwrap.py", line 129, in _expect_inner
    return self.child.expect(expects, timeout=timeout)
  File "/usr/local/lib/python2.7/dist-packages/pexpect/spawnbase.py", line 319, in expect
    compiled_pattern_list = self.compile_pattern_list(pattern)
  File "/usr/local/lib/python2.7/dist-packages/pexpect/spawnbase.py", line 221, in compile_pattern_list
    self._pattern_type_err(p)
  File "/usr/local/lib/python2.7/dist-packages/pexpect/spawnbase.py", line 172, in _pattern_type_err
    for ast in self.allowed_string_types])
TypeError: got <type 'str'> ('\n') as pattern, must be one of: <type 'unicode'>, pexpect.EOF, pexpect.TIMEOUT

**Comment by blink1073** [comment]
Thank you for the heads up, I'll look at this over the weekend.

**Comment by blink1073** [comment]
Hi @lc82111, I can't currently get Scilab to work on my computer, are you able to change the line in `scilab_kernel/kernel.py` to `wrapper.child.linesep = u'\n'` locally?

**Comment by blink1073** [comment]
You could instead try `pip install -U metakernel` instead, I made a fix there in v0.17.1 just now.

**Comment by lc82111** [comment]
I try metakernel 0.17.1,  and issue command "ipython console --kernel scilab"  in term. The above exception gone. But I got following stronger outputs: 
`In [1]: x=1;
      : 
h = gdf();
h.figure_position = [0, 0];
h.figure_size = [560,420];
h.axes_size = [560 * 0.98, 420 * 0.8];
h.visible = "on";
try
getd(".");
catch
end
ids_array=winsid();
for i=1:length(ids_array)
id=ids_array(i);
outfile = sprintf('/tmp/tmpi6a2_6/__ipy_sci_fig_%03d', i);
disp(outfile)
if 'svg' == 'jpg' then
xs2jpg(id, outfile + '.jpg');
elseif 'svg' == 'jpeg' then
xs2jpg(id, outfile + '.jpeg');
elseif 'svg' == 'png' then
xs2png(id, outfile);
else
xs2svg(id, outfile);
end
close(get_figure_handle(id));
end

In [2]: 
`


Blow is my scilab-kernel dependencies:
`scilab-kernel==0.7.1
  - metakernel [required: >=0.10.5, installed: 0.17.1]
    - IPython [required: >=3.0, installed: 5.1.0]
      - backports.shutil-get-terminal-size [required: None, installed: 1.0.0]
      - decorator [required: None, installed: 4.0.10]
      - pathlib2 [required: None, installed: 2.1.0]
        - six [required: None, installed: 1.10.0]
      - pexpect [required: None, installed: 4.2.1]
        - ptyprocess [required: >=0.5, installed: 0.5.1]
      - pickleshare [required: None, installed: 0.7.4]
        - pathlib2 [required: None, installed: 2.1.0]
          - six [required: None, installed: 1.10.0]
      - prompt-toolkit [required: >=1.0.3,<2.0.0, installed: 1.0.9]
        - six [required: >=1.9.0, installed: 1.10.0]
        - wcwidth [required: None, installed: 0.1.7]
      - pygments [required: None, installed: 2.1.3]
      - setuptools [required: >=18.5, installed: 32.3.1]
      - simplegeneric [required: >0.8, installed: 0.8.1]
      - traitlets [required: >=4.2, installed: 4.3.1]
        - decorator [required: None, installed: 4.0.10]
        - enum34 [required: None, installed: 1.1.6]
        - ipython-genutils [required: None, installed: 0.1.0]
        - six [required: None, installed: 1.10.0]
    - pexpect [required: >=4.2, installed: 4.2.1]
      - ptyprocess [required: >=0.5, installed: 0.5.1]
`

**Comment by lc82111** [comment]
By the way, I send scilab server's acount and passwd to steven.silvester@ieee.org. 
Feel free to debug on it.

**Comment by lc82111** [comment]
I also try `wrapper.child.linesep = u'\n'`. It also output 
```
In [1]: x=1;
      : 
h = gdf();
h.figure_position = [0, 0];
h.figure_size = [560,420];
h.axes_size = [560 * 0.98, 420 * 0.8];
h.visible = "on";
try
getd(".");
catch
end
ids_array=winsid();
for i=1:length(ids_array)
id=ids_array(i);
outfile = sprintf('/tmp/tmpfb47CX/__ipy_sci_fig_%03d', i);
disp(outfile)
if 'svg' == 'jpg' then
xs2jpg(id, outfile + '.jpg');
elseif 'svg' == 'jpeg' then
xs2jpg(id, outfile + '.jpeg');
elseif 'svg' == 'png' then
xs2png(id, outfile);
else
xs2svg(id, outfile);
end
close(get_figure_handle(id));
end

In [2]: 






```

**Comment by blink1073** [comment]
Hi @lc82111, I just released version `0.8` which should fix these issues.  `pip install -U scilab_kernel`.

**Comment by lc82111** [comment]
Thanks, I will try it.
